### PR TITLE
add node.go unit test file node_test.go

### PR DESCRIPTION
**Pull request by me020523**
add some unit tests for node.go

### CODE REVIEW

File: trie/node_test.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package trie
```
Comment (holiman) [inline trie/node_test.go]: +license

File: trie/node_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+func newFullNode(v []byte) []interface{} {
```
Comment (rjl493456442) [inline trie/node_test.go]: the `v` is not used here.

File: trie/trie_test.go [hunk 3]
```diff
@@ -546,16 +546,3 @@ func updateString(trie *Trie, k, v string) {
...
-	var (
-		hash  = make([]byte, 20)
-		elems = make([]byte, 20)
-	)
-	for i := 0; i < 5000000; i++ {
-		rand.Read(hash)
-		rand.Read(elems)
-		decodeNode(hash, elems)
-	}
-}
```
Comment (karalabe) [inline trie/trie_test.go]: Why delete this one?

**Comment by holiman** [comment]
Please also add back the deleted test

File: trie/trie_test.go (hunk 3 above)
Comment (me020523) [reply trie/trie_test.go]: this test seems useless, and it tests nothing, so i decide to delete it. already rollback.

File: trie/node_test.go (hunk 1 above)
Comment (me020523) [reply trie/node_test.go]: done

File: trie/node_test.go (hunk 2 above)
Comment (me020523) [reply trie/node_test.go]: already fixed it.

**Pull request merged**